lib/mysql/connector/cursor_cext.py

```diff
@@ -105,7 +105,7 @@ def reset(self, free=True):
         When free is True (default) the result will be freed.
         """
         self._rowcount = -1
-        self._nextrow = None
+        self._nextrow = (None, None)
```

Comment:
This line breaks the fetchone() function. Changing this line doesn't seem to be mandatory though since the fix below still makes fetchmany() work properly...

Reply:
Thank you @paulkned, I didn't consider that self._nextrow is shared with fetchone() function. Line108 is not necessary for a change to fix the fetchmany issue.